File: Darts/src/burlton/dartzee/code/bean/ComboBoxGameType.java
package burlton.dartzee.code.bean;

import burlton.dartzee.code.db.GameEntity;
import burlton.desktopcore.code.bean.ComboBoxItem;

import javax.swing.*;
import java.util.List;

public class ComboBoxGameType extends JComboBox<ComboBoxItem<Integer>>
{
	public ComboBoxGameType()
	{
		DefaultComboBoxModel<ComboBoxItem<Integer>> model = new DefaultComboBoxModel<>();
		
		List<Integer> gameTypes = GameEntity.getAllGameTypes();
		for (int gameType : gameTypes)
		{
			ComboBoxItem<Integer> item = new ComboBoxItem<>(gameType, GameEntity.getTypeDesc(gameType));
			model.addElement(item);
		}
		
		setModel(model);
	}
	
	public int getGameType()
	{
		int ix = getSelectedIndex();
		ComboBoxItem<Integer> selectedItem = getItemAt(ix);
		return selectedItem.getHiddenData();
	}
}

File: Darts/src/burlton/dartzee/code/bean/ComboBoxGameType.kt
package burlton.dartzee.code.bean

import burlton.dartzee.code.utils.getAllGameTypes
import burlton.dartzee.code.utils.getTypeDesc
import burlton.desktopcore.code.bean.ComboBoxItem
import burlton.desktopcore.code.bean.selectedItemTyped
import javax.swing.DefaultComboBoxModel
import javax.swing.JComboBox

class ComboBoxGameType : JComboBox<ComboBoxItem<Int>>()
{
    init
    {
        val model = DefaultComboBoxModel<ComboBoxItem<Int>>()

        val gameTypes = getAllGameTypes()
        for (gameType in gameTypes)
        {
            val item = ComboBoxItem(gameType, getTypeDesc(gameType))
            model.addElement(item)
        }

        setModel(model)
    }

    fun getGameType() = selectedItemTyped().hiddenData
}
